iib/web/models.py

```diff
@@ -1943,3 +1949,133 @@ def get_mutable_keys(self) -> Set[str]:
+                'overwrite_from_index',
+                'binary_image',
+                'build_tags',
+            },
+        )
+
+        # Validate parent_bundle_image is correctly provided
+        fbc_fragment = request_kwargs.get('fbc_fragment')
+        if not isinstance(fbc_fragment, str):
+            raise ValidationError('The "fbc_fragment" must be a string')
+        request_kwargs['fbc_fragment'] = Image.get_or_create(pull_specification=fbc_fragment)
+
+        from_index = request_kwargs.get('from_index', None)
+        if not (isinstance(from_index, str) and from_index):
+            raise ValidationError('The "from_index" value must be a string')
+        request_kwargs['from_index'] = Image.get_or_create(pull_specification=from_index)
+
+        # Validate binary_image is correctly provided
+        binary_image = request_kwargs.pop('binary_image', None)
+        if binary_image is not None and not isinstance(binary_image, str):
+            raise ValidationError('The "binary_image" value must be a string')
+        elif not binary_image and not current_app.config['IIB_BINARY_IMAGE_CONFIG']:
+            raise ValidationError('The "binary_image" value must be a non-empty string')
+
+        # Verify that `overwrite_from_index` is the correct type
+        overwrite = request_kwargs.pop('overwrite_from_index', False)
+        if not isinstance(overwrite, bool):
+            raise ValidationError('The "overwrite_from_index" parameter must be a boolean')
+
+        # Verify that `overwrite_from_index_token` is the correct type
+        overwrite_token = request_kwargs.pop('overwrite_from_index_token', None)
+        if overwrite_token:
+            if not isinstance(overwrite_token, str):
+                raise ValidationError('The "overwrite_from_index_token" parameter must be a string')
+            if overwrite_token and not overwrite:
+                raise ValidationError(
+                    'The "overwrite_from_index" parameter is required when'
+                    ' the "overwrite_from_index_token" parameter is used'
+                )
+
+        # Remove attributes that are not stored in the database
+        request_kwargs.pop('overwrite_from_index_token', None)
+
+        # Validate add_arches are correctly provided
+        add_arches = request_kwargs.pop('add_arches', [])
+        Architecture.validate_architecture_json(add_arches)
+
+        # Verify the user is authorized to use overwrite_from_index
+        # current_user.is_authenticated is only ever False when auth is disabled
+        if current_user.is_authenticated:
+            if overwrite and not overwrite_token:
+                raise Forbidden(
+                    'You must set "overwrite_from_index_token" to use "overwrite_from_index"'
+                )
+
+        # Add the request to a new batch
+        batch = Batch()
+        db.session.add(batch)
+        request_kwargs['batch'] = batch
+
+        request = cls(**request_kwargs)
```

Comment:
For other models we also add,
if current_user.is_authenticated:
            request_kwargs['user'] = current_user
I think we will need this here as well to add the username for the request. I couldn't verify this as we don't have auth enabled on our dev-env, so difficult to test

Reply:
added current_user and distribution_scope